### PR TITLE
Add support for DATE Type in Parquet writer

### PR DESCRIPTION
**Pull request by majetideepak**
Resolves https://github.com/facebookincubator/velox/issues/3918

### CODE REVIEW

**Comment by Yuhta** [comment]
The E2E filter test crashed somehow

**Comment by majetideepak** [comment]
Looks like only the release build is crashing. I will take a look.

**Comment by majetideepak** [comment]
@Yuhta the failure is due to some build quirk that I could not figure out. Just moving the test around fixes the problem.

File: velox/dwio/parquet/tests/reader/E2EFilterTest.cpp [hunk 1]
```diff
@@ -516,6 +516,25 @@ TEST_F(E2EFilterTest, largeMetadata) {
   EXPECT_EQ(1000, reader->numberOfRows());
 }
 
+TEST_F(E2EFilterTest, date) {
```
Comment (Yuhta) [inline velox/dwio/parquet/tests/reader/E2EFilterTest.cpp:519]: Can we copy the same test on DWRF E2E test as well?  Since it's format independent

Comment (majetideepak) [reply velox/dwio/parquet/tests/reader/E2EFilterTest.cpp:519]: DWRF does not support Date type right?

Comment (Yuhta) [reply velox/dwio/parquet/tests/reader/E2EFilterTest.cpp:519]: ORC has it so it probably should be supported.  Just try copy the test there and see if it works, if it is not working then forget about it.

Comment (majetideepak) [reply velox/dwio/parquet/tests/reader/E2EFilterTest.cpp:519]: @Yuhta I see the following error
```
unknown file: Failure
C++ exception with description "Exception: VeloxException
Error Source: RUNTIME
Error Code: UNKNOWN
Reason: not supported yet DATE
Retriable: False
Function: create
File: /Users/deepak/workspace/ahana/velox/velox/dwio/dwrf/writer/ColumnWriter.cpp
Line: 2058
```

File: velox/type/Filter.h [hunk 2]
```diff
@@ -1679,6 +1680,8 @@ static inline bool applyFilter(TFilter& filter, T value) {
     return filter.testInt128(value);
   } else if constexpr (std::is_same_v<T, UnscaledShortDecimal>) {
     return filter.testInt64(value.unscaledValue());
+  } else if constexpr (std::is_same_v<T, Date>) {
```
Comment (Yuhta) [inline velox/type/Filter.h:1683]: Can we add a smoke test to https://github.com/facebookincubator/velox/blob/main/velox/type/tests/FilterTest.cpp?

Comment (majetideepak) [reply velox/type/Filter.h:1740]: done.

**Comment by majetideepak** [comment]
@Yuhta this PR fell through. Can you take a final look? Thanks.

**Pull request closed without merging**